examples/gno.land/p/demo/grc/grc721/grc721_royalty.gno

```diff
@@ -52,20 +52,23 @@ func (r *royaltyNFT) SetTokenRoyalty(tid TokenID, royaltyInfo RoyaltyInfo) error
 	}
 
 	// Set royalty information for the token
-	r.tokenRoyaltyInfo.Set(string(tid), royaltyInfo)
+	r.tokenRoyaltyInfo.Set(tid.String(), royaltyInfo)
 
 	return nil
 }
 
 // RoyaltyInfo returns the royalty information for the given token ID and sale price.
 func (r *royaltyNFT) RoyaltyInfo(tid TokenID, salePrice int64) (std.Address, int64, error) {
 	// Retrieve royalty information for the token
-	val, found := r.tokenRoyaltyInfo.Get(string(tid))
+	val, found := r.tokenRoyaltyInfo.Get(tid.String())
 	if !found {
 		return "", 0, ErrInvalidTokenId
 	}
 
-	royaltyInfo := val.(RoyaltyInfo)
+	royaltyInfo, ok := val.(RoyaltyInfo)
+	if !ok {
+		return "", 0, ErrInvalidRoyaltyInfo
+	}
```

Comment:
The `SetTokenRoyalty()`is making sure that the value is always right, so the ErrInvalidRoyaltyInfo could potentially be removed